1.0.0

alrocar opened the pull request:
Hey, @danicarrion @javisantana 

In my opinion 1.0.0 branch is ready to be merged into master.

What do you think?

BTW, once I have to merge, how should I do it? Seems like both branches won't automatically merge...

EDIT: OK seems like there are only 3 conflicting files... not an issue.

javisantana commented:
About the docs, are we planning to publish it in our website? (Nice to see
sphinx there 😀)

Do we have CI integration?

On Thu, May 11, 2017, 09:49 Alberto Romeu <notifications@github.com> wrote:

> Hey, @danicarrion <https://github.com/danicarrion> @javisantana
> <https://github.com/javisantana>
>
> In my opinion 1.0.0 branch is ready to be merged into master.
>
> What do you think?
>
> BTW, once I have to merge, how should I do it? Seems like both branches
> won't automatically merge...
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/CartoDB/carto-python/pull/52
> Commit Summary
>
>    - total rewrite with some new features
>    - made some edits and tests for NoAuthClient
>    - removed comments and extra code
>    - Merge pull request #20 from CartoDB/r-edits
>    - finished import tests
>    - fixed python sdk func
>    - minor corrections
>    - Merge pull request #23 from CartoDB/r2
>    - working on carto export
>    - working within sdk
>    - simplifying exportjob, revising tests
>    - have a working and polished export job, going to work on export
>    manager
>    - checking that everything works
>    - stopped working on exportManager and exportJob works and is cleaned
>    up
>    - this works on my machiens, will merge with remote
>    - finished export, code polished, git will be up to status, going to
>    work on named maps
>    - got methods working, need to transform to sdk style and then
>    implement manager
>    - made the maps_api more like import_api
>    - finsihed named maps, and the format matches import_api, just need to
>    figure out namedMapManager
>    - got manager to work pretty well
>    - fixed small issues with maps_api file
>    - updated file
>    - fixed small edits in exportJob
>    - Merge pull request #26 from CartoDB/r-export
>    - fixed some pep8 stuff
>    - started work on batch sql api
>    - finishing up batch api functions and dealing with update
>    - finished most of the commands in the batch api, need to talk to dani
>    - fixed namedMapManager and altered namedMaps
>    - finished namedmaps and manager and tests
>    - fixed import_api formatting
>    - fixed formatting
>    - fixed slightly faulty test
>    - trying to get namedmaps mergeable
>    - cleaned up batch api, ready for review
>    - Merge branch '1.0.0' into r-namedmaps
>    - finished last bits of edits
>    - Merge pull request #28 from CartoDB/r-batch
>    - fixed test
>    - pushing namedmaps when merge with everything else
>    - test suuite working
>    - fixed h_method
>    - Merge pull request #27 from CartoDB/r-namedmaps
>    - the python sdk should be fully functional
>    - finished last bit
>    - fixed sqlclient name
>    - Merge pull request #29 from CartoDB/r-docs
>    - made sdk more pythonic
>    - make small edits
>    - no files for testing
>    - Merge pull request #30 from CartoDB/r-docs
>    - massive refactoring, interim commit
>    - user tests pass
>    - comment user tests
>    - dataset tests now pass too (still interim release!!!)
>    - emptied main __init__.py
>    - rough refactor to use pyrestcli
>    - moving everything to the project root
>    - test folder renamed, tests out of setup.py
>    - everything migrated to pyrestcli except maps, sql and exports
>    - visualizations can be exported
>    - added pyrestcli dependency to requirements.txt
>    - CartoException was accidentally gone
>    - Fixed params in constructor call
>    - fixed visualization export state options
>    - minor changes to work with pyrestcli
>    - removing batch manager
>    - fix on sending requests to bath sql api
>    - Merge pull request #33 from CartoDB/remove-sqlbatchmanager
>    - Merge pull request #34 from CartoDB/jsanz-fix-sendsqlbatch
>    - Auth and SQL clients store base_url and username, base_url doesn't
>    expect trailing /api anymore
>    - Enforce trailing / from base_url
>    - Merge pull request #46 from dmed256/1.0.0
>    - added ORGANIZATION variable since it is needed for test fixtures
>    - Added tests for auth.py
>    - Updated README.md with instructions to run tests
>    - excludes carto.com/u/<username> since api endpoints go to <username>.
>    carto.com
>    - Fixed malformed named map template JSON
>    - Fixed named maps APIs
>    - removes unneeded test
>    - added API method to instantiate anonymous maps
>    - Merge pull request #48 from CartoDB/auth-username-fix
>    - moved tests for the SQL API to pytest
>    - updated to pyrestcli 0.6.2. This change breaks compatibility with
>    pyrestcli 0.6.1, so you need to upgrade requirements
>    - Use pyrestcli's new session argument
>    - Merge pull request #51 from dmed256/1.0.0
>    - Merged changes in 1.0.0 related to pyrestcli new session parameter
>    - fixed file import API and added tests
>    - fixes #14
>    - Reverting some tests related to file import API since they were
>    already in the datasets manager tests
>    - ensuring non existing named maps before creation
>    - added myself as a contributor
>    - fixes #32
>    - Added .idea folder to gitignore
>    - Added DS_Store to gitignore
>    - Fixes #38
>    - Added examples from the examples branch
>    - removed old tests
>    - Fixed example description
>    - Change dataset privacy example
>    - Example to export a map as a .carto file
>    - Example for creating a named map
>    - minor fix
>    - Added instantiate map example
>    - removed unused code
>    - Create anonymous map example
>    - Organized examples
>    - Adapted to a change in pyrestcli
>    - Handling exceptions
>    - Added Sphinx API docs
>    - Added API docs in README.md
>    - Updated README.md
>    - Updated requirements and added README.md for the examples folder
>    - Merge branch '1.0.0' into 1.0.0-namedmaps
>    - Update to version 0.6.3 of pyrestcli
>    - Fixed coding style
>    - Added local variables to gitignore
>    - Fixed coding style
>    - Fixed coding style
>    - Fixed coding style
>    - Fixed coding style
>    - Merged changes
>    - Fixed a bug when building maps API urls
>    - Fixed coding style
>    - Added API for downloading map tiles
>    - Merge pull request #49 from CartoDB/1.0.0-namedmaps
>    - Added warnings when using non-public APIs
>    - Updated README.md
>    - Added warnings for non-public APIs in the API docs
>
> File Changes
>
>    - *M* .gitignore
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-0> (4)
>    - *M* CONTRIBUTORS
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-1> (2)
>    - *M* README.md
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-2> (380)
>    - *R* carto/__init__.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-3> (0)
>    - *A* carto/auth.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-4> (106)
>    - *A* carto/datasets.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-5> (198)
>    - *A* carto/exceptions.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-6> (19)
>    - *A* carto/export.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-7> (74)
>    - *A* carto/fields.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-8> (49)
>    - *A* carto/file_import.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-9> (155)
>    - *A* carto/maps.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-10> (236)
>    - *A* carto/paginators.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-11> (49)
>    - *A* carto/permissions.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-12> (44)
>    - *A* carto/resources.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-13> (106)
>    - *A* carto/sql.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-14> (224)
>    - *A* carto/sync_tables.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-15> (152)
>    - *A* carto/tables.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-16> (65)
>    - *A* carto/users.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-17> (109)
>    - *A* carto/visualizations.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-18> (162)
>    - *D* cartodb/__init__.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-19> (4)
>    - *D* cartodb/cartodb.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-20> (278)
>    - *D* cartodb/imports.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-21> (189)
>    - *A* doc/Makefile
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-22> (20)
>    - *A* doc/build-doc.sh
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-23> (7)
>    - *A* doc/source/carto.rst
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-24> (134)
>    - *A* doc/source/conf.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-25> (153)
>    - *A* doc/source/index.rst
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-26> (20)
>    - *A* doc/source/modules.rst
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-27> (7)
>    - *A* examples/README.md
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-28> (21)
>    - *A* examples/change_dataset_privacy.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-29> (59)
>    - *A* examples/check_query.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-30> (77)
>    - *A* examples/create_anonymous_map.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-31> (57)
>    - *A* examples/create_named_map.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-32> (59)
>    - *A* examples/export_map.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-33> (53)
>    - *A* examples/files/anonymous_map.json
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-34> (27)
>    - *A* examples/files/barris_barcelona_1_part_1.csv
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-35> (10)
>    - *A* examples/files/barris_barcelona_1_part_2.csv
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-36> (11)
>    - *A* examples/files/barris_barcelona_1_part_3.csv
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-37> (11)
>    - *A* examples/files/barris_barcelona_1_part_4.csv
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-38> (11)
>    - *A* examples/files/barris_barcelona_1_part_5.csv
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-39> (11)
>    - *A* examples/files/barris_barcelona_1_part_6.csv
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-40> (11)
>    - *A* examples/files/barris_barcelona_1_part_7.csv
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-41> (15)
>    - *A* examples/files/instantiate_map.json
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-42> (3)
>    - *A* examples/files/named_map.json
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-43> (42)
>    - *A* examples/import_and_merge.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-44> (140)
>    - *A* examples/import_standard_table.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-45> (64)
>    - *A* examples/import_sync_table.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-46> (74)
>    - *A* examples/import_sync_table_as_dataset.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-47> (68)
>    - *A* examples/instantiate_named_map.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-48> (65)
>    - *A* examples/kill_query.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-49> (54)
>    - *A* examples/list_tables.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-50> (224)
>    - *A* examples/map_info.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-51> (85)
>    - *A* examples/requirements.txt
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-52> (4)
>    - *A* examples/running_queries.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-53> (56)
>    - *A* examples/sql_batch_api_jobs.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-54> (75)
>    - *A* examples/table_info.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-55> (175)
>    - *A* examples/user_info.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-56> (102)
>    - *M* requirements.txt
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-57> (6)
>    - *D* secret.py.example
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-58> (9)
>    - *M* setup.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-59> (26)
>    - *D* test/client.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-60> (63)
>    - *A* tests/conftest.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-61> (54)
>    - *A* tests/fake.html
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-62> (0)
>    - *A* tests/fake.mp3
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-63> (0)
>    - *A* tests/secret.py.example
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-64> (96)
>    - *R* tests/test.csv
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-65> (0)
>    - *A* tests/test_auth.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-66> (27)
>    - *A* tests/test_datasets.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-67> (96)
>    - *A* tests/test_file_imports.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-68> (25)
>    - *A* tests/test_maps.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-69> (87)
>    - *A* tests/test_sql.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-70> (98)
>    - *A* tests/test_sync_tables.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-71> (25)
>    - *A* tests/test_users.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-72> (81)
>    - *A* tests/test_visualizations.py
>    <https://github.com/CartoDB/carto-python/pull/52/files#diff-73> (32)
>
> Patch Links:
>
>    - https://github.com/CartoDB/carto-python/pull/52.patch
>    - https://github.com/CartoDB/carto-python/pull/52.diff
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CartoDB/carto-python/pull/52>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAB1wxagSbFwe-dfrjiVQp0kDIqe-6Tjks5r4r2CgaJpZM4NXoYX>
> .
>
-- 
-- CARTO CTO

alrocar commented:
About the docs, I think @jsanz proposed publish them at `readthedocs` (at the moment they are just API docs, not developer documentation yet)

There's no CI integration. Should we go with our own CI server or is it OK with something like `travis`?

jsanz commented:
To publish at `readthedocs` we don't need any CI service, just configure the webhook on the repo and that's all. I checked and the only concern is that, as you say, the `doc` folder compiles the API documentation but not the valuable information on the README file in the root. I'll check if there's a way to include it but I wouldn't wait for it for the release.

javisantana commented:
we use travis when possible

danicarrion commented:
I think we should leave CI out of the equation for now. We're not going to be releasing that often, IMHO.

javisantana commented:
I'd not release without CI, that's a min requirement for something
"official" (we are leaving a bunch of things but we can live without them )
and it takes a few hours. It's a little effort and helps a lot when getting
3rd party PR

My 2 cents

On Mon, May 15, 2017, 12:53 Dani Carrión <notifications@github.com> wrote:

> I think we should leave CI out of the equation for now. We're not going to
> be releasing that often, IMHO.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CartoDB/carto-python/pull/52#issuecomment-301442017>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAB1w01HEPOFlm6VEQVFIhdtz35Y9zNKks5r6C6QgaJpZM4NXoYX>
> .
>
-- 
-- CARTO CTO

javisantana commented:
FYI, this is the carto api template doc https://docs.google.com/document/d/14tn8SlFsraodRDcMns5CMYIFsSUs8E3X9OrfrVhCwfY/edit

jsanz commented:
I guess you mean using CI for running the tests, right? If docs are
automatically set up, I don't see any other place where CI can help on this
one.

alrocar commented:
Thanks for the doc, I will go through it to see what's pending.

About CI, configuring a job is easy, the issue is that there are some end to end tests, that make actual API requests (**with an API key**).

Should we use our own CI server instead of Travis?

The other option is to mock all the API requests and use the integration of Github and Travis (not sure if this is easy to do...).

Opinions?

javisantana commented:
I'd disable those tests on Travis if that's the issue (there is env
variables to check if you are running on Travis)

On Mon, May 15, 2017, 13:50 Alberto Romeu <notifications@github.com> wrote:

> Thanks for the doc, I will go through it to see what's pending.
>
> About CI, configuring a job is easy, the issue is that there are some end
> to end tests, that make actual API requests (*with an API key*).
>
> Should we use our own CI server instead of Travis?
>
> The other option is to mock all the API requests and use the integration
> of Github and Travis (not sure if this is easy to do...).
>
> Opinions?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CartoDB/carto-python/pull/52#issuecomment-301452634>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAB1wyes-b7F1ups4lQ2WNBjl6Xnk-J6ks5r6Dv0gaJpZM4NXoYX>
> .
>
-- 
-- CARTO CTO

txomon commented on carto/sql.py:
```diff
@@ -0,0 +1,224 @@
+        self.api_key = self.auth_client.api_key \
+            if hasattr(self.auth_client, "api_key") else None
+        self.base_url = self.auth_client.base_url
+        self.username = self.auth_client.username \
```
this (and the ones around) can be replaced by `getattr(self.auth_client, 'username', None)`

txomon commented on carto/visualizations.py:
```diff
@@ -0,0 +1,162 @@
+        :raise: CartoException
+        """
+        try:
+            if "params" not in client_args:
```
`client_args.setdefault('params', {})`

txomon commented on carto/auth.py:
```diff
@@ -0,0 +1,106 @@
+        """
+        try:
+            http_method = http_method.lower()
+            if (http_method == "post" or http_method == "put") \
```
`http_method in ['post','put']`

txomon commented on CONTRIBUTORS:
```diff
@@ -2,3 +2,5 @@ Javier Santana <jsantana@vizzuality.com>
 Daniel Carrion <daniel@cartodb.com>
 Javier G. Sogo <jgsogo@gmail.com>
+Rohan Sharan <sharanrohan@berkeley.edu>
+Alberto Romeu <alrocar@cartodb.com>
```
No newline at the end of the file

txomon commented on carto/users.py:
```diff
@@ -0,0 +1,109 @@
+    def get_resource_endpoint(self, resource_id):
+        """
+        """
+        return urljoin(self.get_collection_endpoint(), str(resource_id)) \
```
a function for a multiline one liner is maybe too much? I would suggest to split it in (at least) two like:
```
if resource_id is not None:
    return urljoin(self.get_collection_endpoint(), str(resource_id))
```

By default python returns None. Although I would suggest doing it explicitly just in case...

alrocar commented:
Hey @txomon thanks for reviewing. I'll take into account your comments.

txomon commented on setup.py:
```diff
@@ -1,20 +1,10 @@
-      url='https://github.com/Vizzuality/cartodb',
-      install_requires=REQUIRES,
-      packages=['cartodb'],
-      requires = REQUIRES,
```
This line is installation breaking. Currently you don't specify any requirements for your package, you just have a requirements.txt.

When installing things from pip, this line needs to be in place. The appropriate thing to do would be to read from `requirements.txt`.

To test, please have some examples, and create a project with them, using this as a dependency.

you can put this line in the requirements.txt of that examples project to try to reproduce. Also, remember that it should be a clean virtualenv.

`-e git+https://github.com/CartoDB/carto-python.git@3a4c514fcd71c94ddbcede710ecafb3db8d3abcb#egg=carto`

alrocar replied:
No problem, we will add it. Thanks for pointing it out

txomon commented on setup.py:
```diff
@@ -1,20 +1,10 @@
+      author="Daniel Carrión",
+      author_email="daniel@cartodb.com",
+      description="SDK around CartoDB's APIs",
+      version="1.0.0",
```
Given that this is a incredible backwards compatible change, instead of marking this as version v1.0.0, and taking into account that there might be some bugs, I would recommend changing the version to a dev/rc candidate, so that people  don't mistakenly upgrade to it with bugs.

The string I would use would be 1.0.0a1 to mark it as alpha, and not have it upgraded to by pip by default.

More on versioning: https://www.python.org/dev/peps/pep-0440/#examples-of-compliant-version-schemes

More on pip install's mechanism: https://pip.pypa.io/en/stable/reference/pip_install/#pre-release-versions

alrocar commented on carto/auth.py:
```diff
@@ -0,0 +1,106 @@
+    You can find your API key by clicking on the API key section of the user
+    dropdown menu
+    """
+    def __init__(self, base_url, api_key, organization=None, session=None):
```
Let's keep it as `base_url` since changing to `domain` may be confusing for clients not using the cloud version of CARTO.

Thanks!

alrocar commented on carto/maps.py:
```diff
@@ -0,0 +1,236 @@
+        :raise: CartoException
+        """
+        try:
+            if (auth is not None):
```
I use flake8... and I have no warnings in `maps.py`. I'll take a look anyway

alrocar commented on carto/auth.py:
```diff
@@ -0,0 +1,106 @@
+
+from .exceptions import CartoException
+
+if sys.version_info >= (3, 0):
```
Maybe `six` is handier, but since we are doing minor checks, let's keep with `sys` to avoid adding another dependency

txomon commented on carto/users.py:
```diff
@@ -105,5 +107,7 @@ def get_collection_endpoint(self):
         """
-        return urljoin(self.get_collection_endpoint(), str(resource_id)) \
-            if resource_id is not None else None
+        if resource_id is not None:
```
LPT: Even shorter, negated if clause

```if resource_id is None:
    return None
return urljoin(self.get_collection_endpoint(), str(resource_id))
```

txomon commented on tests/conftest.py:
```diff
@@ -1,12 +1,26 @@
+if "API_KEY" in os.environ:
+    API_KEY = os.environ["API_KEY"]
+else:
+    from secret import API_KEY
```
The use of python modules that are not in the repo is not a good pattern.

My advice would be to either load them from a json (it's cleaner), or have the tests fail if the environment is not set (the best option really).

txomon commented on .travis.yml:
```diff
@@ -0,0 +1,12 @@
+language: python
+python:
+- '2.7'
+- '3.6'
```
I would recommend adding 3.5 and 3.4 too, there are low level implementation details that make a difference in some pieces of code, and that way you are covered.

alrocar commented:
I've taken some time to move forward this PR:

- Added developer docs (see `doc` folder) and updated README.md
- Added a job in Travis for CI
- Added a couple more of examples
- Some other fixes (thanks @txomon for reviewing)

See [here](https://github.com/CartoDB/solutions/issues/510#issue-176377513) the list of done and pending tasks

Comments? @javisantana @danicarrion

andy-esch commented on carto/auth.py:
```diff
@@ -0,0 +1,108 @@
+            if netloc.startswith('www.'):
+                netloc = netloc.split('www.')[1]
+            m = re.search('^(.*?)\..*', netloc)
+        return m.group(1)
```
This might be a good place to check if `m` is None-valued or not. If it is None-valued, `m.group(1)` throws an error that a user would have to  figure out means the base_url is not in the proper form.

alrocar replied:
Hey Andy,

thanks for reviewing!

It makes sense, I'm adding a fix for it ;)

danicarrion commented:
:heart_eyes:

andy-esch commented:
🎉 🎉 🎉
